File: app/buildz/tmp/kapt3/stubs/debug/com/bits/bee/bpmc/presentation/ui/sign_up/atur_produk/ItemDummyAdapter.java
package com.bits.bee.bpmc.presentation.ui.sign_up.atur_produk;

import java.lang.System;

/**
 * Created by aldi on 31/08/22.
 */
@kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000,\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\u0002\n\u0002\b\u0002\n\u0002\u0010\b\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0004\u0018\u00002\u000e\u0012\u0004\u0012\u00020\u0002\u0012\u0004\u0012\u00020\u00030\u0001:\u0002\u000e\u000fB\u0005\u00a2\u0006\u0002\u0010\u0004J\u0018\u0010\u0005\u001a\u00020\u00062\u0006\u0010\u0007\u001a\u00020\u00032\u0006\u0010\b\u001a\u00020\tH\u0016J\u0018\u0010\n\u001a\u00020\u00032\u0006\u0010\u000b\u001a\u00020\f2\u0006\u0010\r\u001a\u00020\tH\u0016\u00a8\u0006\u0010"}, d2 = {"Lcom/bits/bee/bpmc/presentation/ui/sign_up/atur_produk/ItemDummyAdapter;", "Landroidx/recyclerview/widget/ListAdapter;", "Lcom/bits/bee/bpmc/domain/model/ItemDummy;", "Landroidx/recyclerview/widget/RecyclerView$ViewHolder;", "()V", "onBindViewHolder", "", "holder", "position", "", "onCreateViewHolder", "parent", "Landroid/view/ViewGroup;", "viewType", "DiffCallback", "ViewHolder", "app_debug"})
public final class ItemDummyAdapter extends androidx.recyclerview.widget.ListAdapter<com.bits.bee.bpmc.domain.model.ItemDummy, androidx.recyclerview.widget.RecyclerView.ViewHolder> {
    
    public ItemDummyAdapter() {
        super(null);
    }
    
    @org.jetbrains.annotations.NotNull()
    @java.lang.Override()
    public androidx.recyclerview.widget.RecyclerView.ViewHolder onCreateViewHolder(@org.jetbrains.annotations.NotNull()
    android.view.ViewGroup parent, int viewType) {
        return null;
    }
    
    @java.lang.Override()
    public void onBindViewHolder(@org.jetbrains.annotations.NotNull()
    androidx.recyclerview.widget.RecyclerView.ViewHolder holder, int position) {
    }
    
    @kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000\u001e\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\b\u0086\u0004\u0018\u00002\u00020\u0001B\r\u0012\u0006\u0010\u0002\u001a\u00020\u0003\u00a2\u0006\u0002\u0010\u0004J\u000e\u0010\u0005\u001a\u00020\u00062\u0006\u0010\u0007\u001a\u00020\bR\u000e\u0010\u0002\u001a\u00020\u0003X\u0082\u0004\u00a2\u0006\u0002\n\u0000\u00a8\u0006\t"}, d2 = {"Lcom/bits/bee/bpmc/presentation/ui/sign_up/atur_produk/ItemDummyAdapter$ViewHolder;", "Landroidx/recyclerview/widget/RecyclerView$ViewHolder;", "binding", "Lcom/bits/bee/bpmc/databinding/ItemProdukDummyBinding;", "(Lcom/bits/bee/bpmc/presentation/ui/sign_up/atur_produk/ItemDummyAdapter;Lcom/bits/bee/bpmc/databinding/ItemProdukDummyBinding;)V", "bind", "", "model", "Lcom/bits/bee/bpmc/domain/model/ItemDummy;", "app_debug"})
    public final class ViewHolder extends androidx.recyclerview.widget.RecyclerView.ViewHolder {
        private final com.bits.bee.bpmc.databinding.ItemProdukDummyBinding binding = null;
        
        public ViewHolder(@org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.databinding.ItemProdukDummyBinding binding) {
            super(null);
        }
        
        public final void bind(@org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.domain.model.ItemDummy model) {
        }
    }
    
    @kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000\u0018\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\u000b\n\u0002\b\u0004\u0018\u00002\b\u0012\u0004\u0012\u00020\u00020\u0001B\u0005\u00a2\u0006\u0002\u0010\u0003J\u0018\u0010\u0004\u001a\u00020\u00052\u0006\u0010\u0006\u001a\u00020\u00022\u0006\u0010\u0007\u001a\u00020\u0002H\u0016J\u0018\u0010\b\u001a\u00020\u00052\u0006\u0010\u0006\u001a\u00020\u00022\u0006\u0010\u0007\u001a\u00020\u0002H\u0016\u00a8\u0006\t"}, d2 = {"Lcom/bits/bee/bpmc/presentation/ui/sign_up/atur_produk/ItemDummyAdapter$DiffCallback;", "Landroidx/recyclerview/widget/DiffUtil$ItemCallback;", "Lcom/bits/bee/bpmc/domain/model/ItemDummy;", "()V", "areContentsTheSame", "", "oldItem", "newItem", "areItemsTheSame", "app_debug"})
    public static final class DiffCallback extends androidx.recyclerview.widget.DiffUtil.ItemCallback<com.bits.bee.bpmc.domain.model.ItemDummy> {
        
        public DiffCallback() {
            super();
        }
        
        @java.lang.Override()
        public boolean areItemsTheSame(@org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.domain.model.ItemDummy oldItem, @org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.domain.model.ItemDummy newItem) {
            return false;
        }
        
        @java.lang.Override()
        public boolean areContentsTheSame(@org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.domain.model.ItemDummy oldItem, @org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.domain.model.ItemDummy newItem) {
            return false;
        }
    }
}

File: app/src/main/java/com/bits/bee/bpmc/presentation/ui/sign_up/atur_produk/ItemDummyAdapter.kt
package com.bits.bee.bpmc.presentation.ui.sign_up.atur_produk

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.view.isVisible
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.bits.bee.bpmc.R
import com.bits.bee.bpmc.databinding.ItemProdukDummyBinding
import com.bits.bee.bpmc.domain.model.ItemDummy
import com.bits.bee.bpmc.utils.CurrencyUtils
import com.bits.bee.bpmc.utils.FileHandlerUtils
import com.bits.bee.bpmc.utils.ImageUtils
import java.math.BigDecimal


/**
 * Created by aldi on 31/08/22.
 */
class ItemDummyAdapter(
    private val onItemClick : (ItemDummy) -> Unit
) : ListAdapter<ItemDummy, RecyclerView.ViewHolder>(DiffCallback()) {

    private val VIEW_ITEM = 1
    private val VIEW_ADD = 0

    override fun submitList(list: MutableList<ItemDummy>?) {
        val newList : MutableList<ItemDummy> = mutableListOf()
        newList.addAll(list ?: mutableListOf())
        if(newList.size < 10){
            newList.add(ItemDummy())
        }
        super.submitList(newList)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return  ViewHolder(
            ItemProdukDummyBinding.inflate(inflater, parent, false)
        )
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val viewHolder = holder as ViewHolder
        viewHolder.bind(getItem(position))
        viewHolder.itemView.setOnClickListener {
            onItemClick(getItem(position))
        }
    }

    override fun getItemViewType(position: Int): Int {
        return when(position == currentList.size - 1){
            true -> VIEW_ITEM
            false -> VIEW_ADD
        }
    }

    inner class ViewHolder(private val binding : ItemProdukDummyBinding) : RecyclerView.ViewHolder(binding.root){

        fun bind(model : ItemDummy){
            binding.apply {
                val context = binding.root.context

                tvPid.isVisible = model.id != null
                ivPencil.isVisible = model.id != null
                tvHarga.isVisible = model.id != null

                model.id?.let {
                    tvNamaItem.text = model.name
                    tvPid.isVisible = model.pid.isNotEmpty()
                    tvPid.text = model.pid
                    if (model.price.isEmpty()){
                        tvHarga.text = "0"
                    }else{
                        tvHarga.text = context.getString(R.string.mata_uang_nominal, "Rp.", CurrencyUtils.formatCurrency(
                            BigDecimal(model.price.replace(",", "") ?: "0")
                        ))
                    }
                    if (model.picPath.isNotEmpty()){
                        imageItem.setImageBitmap(FileHandlerUtils.checkDirPath(model.picPath))
                    }else{
                        imageItem.setImageDrawable(ImageUtils.generateFromInitial(context, model.name))
                    }
                } ?: run {
                    tvNamaItem.text = context.getString(R.string.tambah_baru)
                    imageItem.setImageDrawable(ImageUtils.generateFromInitial(context, "+"))
                }
            }
        }
    }


    class DiffCallback() : DiffUtil.ItemCallback<ItemDummy>(){
        override fun areItemsTheSame(oldItem: ItemDummy, newItem: ItemDummy): Boolean {
            return oldItem.id == newItem.id
        }

        override fun areContentsTheSame(oldItem: ItemDummy, newItem: ItemDummy): Boolean {
            return oldItem == newItem
        }

    }
}